recipes/prepare_upgrade.rb

```diff
@@ -54,4 +54,34 @@
         :expat_dir => expat_dir
     })
     action :create
+end
+
+# move base image 
+image_url = node['hops']['docker']['base']['download_url']
+base_filename = File.basename(image_url)
+download_command = " wget #{image_url}"
+
+if node['install']['enterprise']['install'].casecmp? "true"
+  image_url ="#{node['install']['enterprise']['download_url']}/docker-tars/#{node['hops']['docker_img_version']}/#{base_filename}"
+  download_command = " wget --user #{node['install']['enterprise']['username']} --password #{node['install']['enterprise']['password']} #{image_url}"
+end
+
+bash "download_images" do
+  user "root"
+  group "root"
+  sensitive true
+  code <<-EOF
+       #{download_command} -O #{Chef::Config['file_cache_path']}/#{base_filename}
```

Comment:
Download them directly to `#{node['cloud']['init']['install_dir']}` This way the following block will be skipped.

As we discussed offline this block normally won't run as the images must already exist from hops::docker_registry recipe

Reply:
Yeah, i'm fixing that already here 
https://github.com/logicalclocks/hops-hadoop-chef/pull/297
So, we would need both blocks depending on which recipe would first to download the image, and the second block will be responsible for copying